From a769e40c4d7681c8a529c1daa42f38157991886b Mon Sep 17 00:00:00 2001
From: chenj <chenj@lemote.com>
Date: Thu, 1 Dec 2016 09:51:35 +0800
Subject: [PATCH 245/316] AOSCOS: snd-hda-codec: new symbol
 snd_hda_codec_exec_verb

snd_hda_codec_exec_verb() wraps&exports codec_exec_verb()

Signed-off-by: Huacai Chen <chenhc@lemote.com>
Signed-off-by: Hongliang Tao <taohl@lemote.com>
Signed-off-by: Hua Yan <yanh@lemote.com>

Signed-off-by: chenj <chenj@lemote.com>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 include/sound/hda_codec.h |  2 ++
 sound/pci/hda/hda_codec.c | 18 ++++++++++++++++++
 2 files changed, 20 insertions(+)

diff --git a/include/sound/hda_codec.h b/include/sound/hda_codec.h
index 575e55aa08ca..e902583ad41c 100644
--- a/include/sound/hda_codec.h
+++ b/include/sound/hda_codec.h
@@ -368,6 +368,8 @@ struct hda_verb {
 	u32 param;
 };
 
+int snd_hda_codec_exec_verb(struct hda_codec *codec, unsigned int cmd,
+			    unsigned int *res);
 void snd_hda_sequence_write(struct hda_codec *codec,
 			    const struct hda_verb *seq);
 
diff --git a/sound/pci/hda/hda_codec.c b/sound/pci/hda/hda_codec.c
index 14763c0f31ad..cc1447c1dc6c 100644
--- a/sound/pci/hda/hda_codec.c
+++ b/sound/pci/hda/hda_codec.c
@@ -68,6 +68,24 @@ static int codec_exec_verb(struct hdac_device *dev, unsigned int cmd,
 	return err;
 }
 
+int snd_hda_codec_exec_verb(struct hda_codec *codec, unsigned int raw_cmd,
+			   unsigned int *res)
+{
+	unsigned cmd;
+	unsigned int _res;
+	int r;
+
+	cmd = (u32)codec->addr << 28;
+	cmd |= raw_cmd;
+
+	r = codec_exec_verb(&codec->core, cmd, 0, (res || codec->bus->core.sync_write) ? &_res : NULL);
+	if (res)
+		*res = _res;
+
+	return r;
+}
+EXPORT_SYMBOL_GPL(snd_hda_codec_exec_verb);
+
 /**
  * snd_hda_sequence_write - sequence writes
  * @codec: the HDA codec
-- 
2.48.1

